pkg/target/cluster.go

```diff
@@ -17,8 +19,11 @@ import (
 type Cluster struct {
 	Name string
 	// Are ordered by index
-	Hosts []*Host
-	Type  string
+	Hosts                  []*Host
+	AvailableHosts         []*Host
+	Hm                     sync.RWMutex
+	Type                   string
+	UpdateHostHealthStatus chan *HostStatus
```

Comment:
can we do without this?

Reply:
This is how we can simplify. 
* remove the channel for updates to each host
* have atomic `Alive` variable in each `Host`.
  * this variable is set and maintained by the host
  * contention is low because it is changed by the host rarely and read every 10s (see below)
* re-check host statuses in the cluster periodically, say each 10s
* update hosts list accordingly
  * this will introduce avg 5s lb lag, which is ok
  * the operation is very fast and will not block
  * the only block for streaming will be locking during the hosts list update. it is small so it will be fast

this solves two problems:
* decreases complexity
* simplifies the concurrency by *a lot*
* removes feedback from the child (host) to the parent (cluster) in the architectural hierarchy. which poses a design problem

@avereha @gksinghjsr feedback?

Reply:
let's leave this aside for now.